functions/common/config-common/src/main/java/org/springframework/cloud/fn/common/config/CustomizationDoc.java

```diff
@@ -16,23 +16,18 @@
 
 package org.springframework.cloud.fn.common.config;
 
-import java.lang.annotation.Documented;
 import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
- * The annotation to mark a {@code @Bean} method for end user-provided customization
- * via {@link ComponentCustomizer} with this bean type.
+ * The annotation to mark a {@code @Bean} method which provides a
+ * {@link ComponentCustomizer} information for generated documentation.
  *
  * @author Artem Bilan
  *
  * @since 1.2.1
  */
-@Target({ ElementType.METHOD, ElementType.ANNOTATION_TYPE })
-@Retention(RetentionPolicy.RUNTIME)
-@Documented
-public @interface CustomizationAware {
+@Target(ElementType.METHOD)
+public @interface CustomizationDoc {
```

Comment:
Do we envision ever using this annotation to signal something besides generating a doc? If so, then the name `CustomizationDoc` may be limiting.

Reply:
Also, since we are not doing any docs in this PR and we no longer need this annotation I would be in favor of removing this from the current proposal and adding what is needed when we do the docs work.

Reply:
Oh no, I think this is a great simplifying improvement. The fact that we have to mark the component anyways makes total sense to update the signature to take optional customizer bean. It also clearly shows there is only 1 customizer bean that can affect the component. 

I was suggesting we can wait to add the annotation until the docs story though.

Reply:
No problem! Let's hear from others then!
@sobychacko , @corneil .

BTW, I modified JavaDocs as well to reflect reality.
So, I'm not sure what is the concern so far...

Reply:
> BTW, I modified JavaDocs as well to reflect reality.

Yes, the simplification of what was needed to be doc'ed now to explain the feature is also a signal of how this has simplified things. 

> So, I'm not sure what is the concern so far...

I'm not following you on this. Can you clarify?

Reply:
I mean that annotation is now for only docs - nothing more.
I'm not sure that can figure out something else to infer the bean method info for docs generation...

But as you suggest I can remove it for the future consideration.
Not a big deal.

Reply:
Another option would be finding any bean that took a nullable componentcustomizer in its bean method and use that to generate the doc for it. The downside w/ that approach is  that we would have limited context to actually doc whereas w/ an annotation we could add attributes to drive the docs etc.. 

I think we will end up w/ the annotation but we could wait to add it in as part of #275 and not have to think/decide about it in this PR.

Reply:
That sounds like a plan to me.